pkg/controllers/drift/controller.go

```diff
@@ -0,0 +1,137 @@
+			return reconcile.Result{}, fmt.Errorf("patching node, %w", err)
+		}
+	}
+
+	return reconcile.Result{RequeueAfter: 30 * time.Minute}, nil
+}
+
+func (d Drift) Builder(ctx context.Context, m manager.Manager) operatorcontroller.Builder {
+	builder := controllerruntime.
+		NewControllerManagedBy(m).
+		Named(controllerName).
+		For(&v1.Node{}).
+		Watches(
+			&source.Kind{Type: &v1alpha5.Provisioner{}},
+			handler.EnqueueRequestsFromMapFunc(func(o client.Object) (requests []reconcile.Request) {
+				provisioner := o.(*v1alpha5.Provisioner)
+				if provisioner.Spec.ProviderRef == nil {
+					return requests
+				}
+				// Ensure provisioner has a defined AWSNodeTemplate
+				nodeTemplate := &v1alpha1.AWSNodeTemplate{}
+				if err := d.kubeClient.Get(ctx, types.NamespacedName{Name: provisioner.Spec.ProviderRef.Name}, nodeTemplate); err != nil {
+					logging.FromContext(ctx).Errorf("getting AWSNodeTemplates when mapping drift watch events, %s", err)
+					return requests
+				}
+				nodes := &v1.NodeList{}
+				if err := d.kubeClient.List(ctx, nodes, client.MatchingLabels(map[string]string{v1alpha5.ProvisionerNameLabelKey: provisioner.Name})); err != nil {
+					logging.FromContext(ctx).Errorf("listing nodes when mapping drift watch events, %s", err)
+					return requests
+				}
+				for _, node := range nodes.Items {
+					requests = append(requests, reconcile.Request{NamespacedName: types.NamespacedName{Name: node.Name}})
+				}
+				return requests
+			})).
+		WithOptions(controller.Options{MaxConcurrentReconciles: 10})
+
+	if err := m.GetFieldIndexer().IndexField(ctx, &v1alpha5.Provisioner{}, ".spec.providerRef.name", func(rawObj client.Object) []string {
+		provisioner := rawObj.(*v1alpha5.Provisioner)
+		return []string{provisioner.Spec.ProviderRef.Name}
+	}); err != nil {
+		//Return early, controller won't be able to get provisioners related to AWSNodeTemplate if the index field failed
+		logging.FromContext(ctx).Errorf("creating index for provisioner while building drift controller, %w", err)
+		return builder
+	}
+
+	return builder.Watches(
+		&source.Kind{Type: &v1alpha1.AWSNodeTemplate{}},
+		handler.EnqueueRequestsFromMapFunc(func(o client.Object) (requests []reconcile.Request) {
+			provisioners := &v1alpha5.ProvisionerList{}
+			if err := d.kubeClient.List(ctx, provisioners, client.MatchingFields{".spec.providerRef.name": o.GetName()}); err != nil {
+				logging.FromContext(ctx).Errorf("listing provisioners for AWSNodeTemplate reconciliation %w", err)
+				return requests
+			}
+			for _, provisioner := range provisioners.Items {
+				nodes := &v1.NodeList{}
+				if err := d.kubeClient.List(ctx, nodes, client.MatchingLabels(map[string]string{v1alpha5.ProvisionerNameLabelKey: provisioner.Name})); err != nil {
+					logging.FromContext(ctx).Errorf("listing nodes when mapping drift watch events, %s", err)
+					return requests
+				}
+				for _, node := range nodes.Items {
```

Comment:
There's probably ways to abstract this since I see you doing this twice in both `Enqueues...`

Reply:
Sure, I can create a function in the controller. Does that sound good ?

Reply:
Done. Created a function `getReconcileRequest` which is being used in both `Enqueues...`